controller/share_manager_controller.go

```diff
@@ -798,6 +860,40 @@ func (c *ShareManagerController) createShareManagerPod(sm *longhorn.ShareManager
 		return nil, err
 	}
 
+	var affinity *corev1.Affinity
+
+	if pv.Spec.StorageClassName != "" {
+		sc, err := c.ds.GetStorageClass(pv.Spec.StorageClassName)
+		if err != nil {
+			c.logger.WithError(err).Warnf("Failed to get storage class %v", pv.Spec.StorageClassName)
+		} else {
+			affinity = c.getAffinityFromStorageClass(sc)
+
+			// Find the node selector from the storage class and merge it with the system managed components node selector
+			if nodeSelector == nil {
+				nodeSelector = map[string]string{}
+			}
+
+			affinity = c.getAffinityFromStorageClass(sc)
+
+			// Find the node selector from the storage class and merge it with the system managed components node selector
+			if nodeSelector == nil {
+				nodeSelector = map[string]string{}
+			}
+			nodeSelectorFromStorageClass := c.getShareManagerNodeSelectorFromStorageClass(sc)
+			for k, v := range nodeSelectorFromStorageClass {
+				nodeSelector[k] = v
+			}
```

Comment:
This merge means that the final scheduling should satisfy both `node selector from SC` and `system managed components node selector`, right?

Reply:
You are right!